### PR TITLE
C#: Fix qhelp for ICryptoTransform queries

### PR DESCRIPTION
**Pull request by hvitved**

### CODE REVIEW

File: csharp/ql/src/Likely Bugs/ThreadUnsafeICryptoTransformLambda.qhelp [hunk 1]
```diff
@@ -3,8 +3,14 @@
   "qhelp.dtd">
 <qhelp>
 <overview>
-  <include src="ThreadUnsafeICryptoTransform.qhelp" />
-  
+<p>Classes that implement <code>System.Security.Cryptography.ICryptoTransform</code> are not thread safe.</p>
```
Comment (aibaars) [inline csharp/ql/src/Likely Bugs/ThreadUnsafeICryptoTransformLambda.qhelp]: You should be able to extract the shared blob of text using a fragment. See https://github.com/Semmle/ql/blob/22294091808d3af2418ecc3d53b8843eb274277d/java/ql/src/DeadCode/DeadMethod.qhelp#L10 for an example.

Comment (hvitved) [reply csharp/ql/src/Likely Bugs/ThreadUnsafeICryptoTransformLambda.qhelp]: Yeah, that is much nicer. Didn't know that was possible, thanks.

Comment (aibaars) [reply csharp/ql/src/Likely Bugs/ThreadUnsafeICryptoTransformLambda.qhelp]: I wasn't sure either until 15 minutes ago ;-)

**Comment by jf205** [comment]
What was the problem with the qhelp here? (So that I know what to manually look for in future PRs).

**Comment by hvitved** [comment]
> What was the problem with the qhelp here? (So that I know what to manually look for in future PRs).

The problem was that the PR check `Pull-Request-Check/QHelp-Preview` did not run because it was a PR authored outside of Semmle. I believe there will be a fix so externally contributed PRs cannot be merged before such checks have run.

**Comment by jf205** [comment]
Ok. Was the qhelp file itself also faulty though?

**Comment by hvitved** [comment]
> Ok. Was the qhelp file itself also faulty though?

Yes, it was. And the check would have caught it.

**Comment by jf205** [comment]
The question I should have asked to start off with: What is wrong with the qhelp file? :)

**Comment by hvitved** [comment]
> The question I should have asked to start off with: What is wrong with the qhelp file? :)

:-) The problem was this inclusion: https://github.com/Semmle/ql/pull/1167/files#diff-390612a99a79249505d81d52afede629L6.

File: csharp/ql/src/Likely Bugs/ThreadUnsafeICryptoTransform.qhelp [hunk 2]
```diff
@@ -3,15 +3,7 @@
...
-<p>Allowing the same hash object to be called with data from multiple threads before calling the finish function could potentially lead to incorrect results.</p>
-<p>For example, if you have multiple threads hashing <code>"abc"</code> on a static hash object, you may occasionally obtain the results (incorrectly) for hashing <code>"abcabc"</code>, or face other unexpected behavior.</p>
-<p>It is very unlikely somebody outside Microsoft would write a class that implements <code>ICryptoTransform</code>, and even if they do, it is likely that they will follow the same common pattern as the existing classes implementing this interface.</p>
-<p>Any object that implements <code>System.Security.Cryptography.ICryptoTransform</code> should not be used in concurrent threads as the instance members of such object are also not thread safe.</p>
-<p>Potential problems may not be evident at first, but can range from explicit errors such as exceptions, to incorrect results when sharing an instance of such an object in multiple threads.</p>
-
+<include src="ThreadUnsafeICryptoTransformOverview.qhelp" />
```
Comment (aibaars) [inline csharp/ql/src/Likely Bugs/ThreadUnsafeICryptoTransform.qhelp]: Perhaps indent a little. Do we have a qhelp formatter btw?

**Pull request merged**